src/backend/joanie/payment/api.py

```diff
@@ -86,3 +108,65 @@ def tokenize_card(self, request):
         payment_infos = payment_backend.tokenize_card(user=self.request.user)
 
         return Response(payment_infos, status=HTTPStatus.OK)
+
+    @action(
+        methods=["patch"],
+        detail=True,
+    )
+    def promote(self, request, pk=None):  # pylint:disable=unused-argument
+        """
+        Promote the credit card ownership of a user to main card.
+        """
+        credit_card = self.get_object()
+        username = (
+            self.request.auth["username"]
+            if self.request.auth
+            else self.request.user.username
+        )
+        card_ownership = credit_card.ownerships.get(owner__username=username)
+        card_ownership.is_main = True
+        card_ownership.save()
+
+        return Response(HTTPStatus.OK)
+
+    def destroy(self, request, *args, **kwargs):
+        """
+        Delete the relation between the card and the owner when there are many
+        owners on a shared card only if there are no pending payment on the owner's orders.
+        If there is only one owner on the card and there are no pending payments we can delete
+        the credit card.
+        """
+        credit_card = self.get_object()
+        username = (
+            self.request.auth["username"]
+            if self.request.auth
+            else self.request.user.username
+        )
+
+        # Check for pending payments for this user
+        has_pending_payments = Order.objects.filter(
+            owner__username=username,
+            state__in=[
+                ORDER_STATE_PENDING,
+                ORDER_STATE_PENDING_PAYMENT,
+            ],
+            payment_schedule__contains=[{"state": PAYMENT_STATE_PENDING}],
+        ).exists()
```

Comment:
With my current work on Richie, I realized there is something wrong with this query. Furthermore with the recent development of @kernicPanel, we can now assume that if an order is linked to a credit card that means it is required.
```suggestion
        has_pending_payments = credit_card.orders.filter(owner__username=username).exists()
```